src/main/scala/db/UserDB.scala

```diff
@@ -2,47 +2,62 @@ package db
 
 import Dao.DAO
 import models.Users
-import java.util.UUID
-import scala.collection.mutable.ListBuffer
+
+import scala.util.{Failure, Success, Try}
+import java.sql.{ResultSet, SQLException, Statement}
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 class UserDB extends DAO {
 
-  private var userStorage: ListBuffer[Users] = ListBuffer.empty
+  private val statement: Statement = Connection.connection().createStatement()
 
   def addUser(user: Users): Future[String] = Future {
-    userStorage += user
-    Thread.sleep(150)
-    "new user added."
+    val query = s"INSERT INTO userTable (userName, age, address, dateOfBirth, userType) VALUES ('${user.userName}', ${user.age}, '${user.address}', '${user.dateOfBirth}', '${user.userType}') ;"
+
+    Try(statement.executeQuery(query)) match {
+      case Success(_) => "Data inserted successfully!"
+      case Failure(exception) => exception.getMessage
+    }
   }
 
-  def getById(userId: UUID): Future[Option[Users]] = Future {
+  def getById(userId: Int): Future[ResultSet] = Future {
     Thread.sleep(150)
-    userStorage.find(_.userId == userId)
+    val query = s"SELECT  * FROM userTable WHERE userId = $userId ;"
+    Try(statement.executeQuery(query)) match {
+      case Success(resultSet) => resultSet
+      case Failure(_) => throw new SQLException
+    }
   }
 
-  def getAll: Future[ListBuffer[Users]] = Future {
+  def getAll: Future[ResultSet] = Future {
+    val query = "SELECT * FROM userTable;"
     Thread.sleep(150)
-    userStorage
+    Try(statement.executeQuery(query)) match {
```

Comment:
Since this code is duplicated from line 27, move this to its own method and call that method here and on line 27.